src/lib/core/CASEAuthTag.h

```diff
@@ -0,0 +1,43 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <lib/core/CHIPConfig.h>
+#include <lib/support/CodeUtils.h>
+#include <stdlib.h>
+
+namespace chip {
+
+typedef uint32_t CASEAuthTag;
+
+static constexpr CASEAuthTag kUndefinedCAT = 0;
+
+// Muximum number of CASE Authenticated Tags (CAT) in the CHIP certificate subject.
+static constexpr size_t kMaxSubjectCATAttributeCount = CHIP_CONFIG_CERT_MAX_RDN_ATTRIBUTES - 2;
+
+struct CATValues
+{
+    CASEAuthTag val[kMaxSubjectCATAttributeCount];
+
+    size_t size() const { return ArraySize(val); }
+};
+
+static constexpr CATValues kUndefinedCATs = { { kUndefinedCAT } };
```

Comment:
Should we also make the struct CATValues set the values to undefined when it is created?

Reply:
`CATValues` is a `struct` so cannot set a default value.

Reply:
I'm not sure what you mean? You would do it in a constructor, or even by assigning the values to the members inside the struct (which has the same effect, as if a constructor had been provided).

Reply:
Something like:

struct CATValues
{
    CASEAuthTag values[kMaxSubjectCATAttributeCount] **= kUndefinedCATs**;
    // ... more stuff ...
};

Reply:
(Notwithstanding the circular dependency, but zero works also.)

Reply:
Introduction of a constructor or a default value makes it more difficult to work with this struct. specifically, to make an array initialization. So I just verified that we do set a default value everywhere it is used.

Reply:
I believe default value or default constructor does not prevent usage in arrays.

E.g. godbolt.org is fine with:

struct Bar {
    int a = 0;
    int b = 0;
    int c[3] = {0};
};

Bar array[42];

Reply:
Yes, I updated the implementation to set `values` array to the default value. Sorry for confusion...